javalin/src/main/java/io/javalin/compression/CompressedStream.kt

```diff
@@ -35,31 +30,15 @@ internal class CompressedOutputStream(val compression: CompressionStrategy, val
 
 }
 
-private fun tryMatchCompression(compression: CompressionStrategy, ctx: Context, originStream: OutputStream): Pair<CompressionType, OutputStream>? =
+private fun tryMatchCompression(
+    compression: CompressionStrategy,
+    ctx: Context,
+): Compressor? =
     ctx.header(Header.ACCEPT_ENCODING)?.let { acceptedEncoding ->
-        tryBrotli(compression, originStream, acceptedEncoding) ?: tryGzip(compression, originStream, acceptedEncoding)
+        compression.compressors.firstOrNull { acceptedEncoding.contains(it.type().typeName, ignoreCase = true) }
```

Comment:
It's common for clients to list all of their supported encodings. Is priority handled by insertion order? Should we make it possible to specify priority?

Reply:
Priority **is** handled by insertion order, since the configuration (abstraction) won't change I didn't give the possibility to prioritize. I respected the current order of `brotli -> gzip -> none`.

Reply:
Yes, I'm asking if we should make it possible.

Reply:
> Yes, I'm asking if we should make it possible. 

I don't think it's worth doing before 6.x with the removal of optional dependencies.